components/partners/bhaskar/src/main/java/org/mozilla/rocket/bhaskar/Repository.java

```diff
@@ -0,0 +1,182 @@
+import android.content.Context;
+import android.support.annotation.Nullable;
+import android.support.v4.util.Pair;
+import android.text.TextUtils;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.mozilla.cachedrequestloader.CachedRequestLoader;
+import org.mozilla.cachedrequestloader.ResponseData;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+
+public class Repository {
+    private int currentPage = FIRST_PAGE;
+    private PageSubscription currentPageSubscription;
+    private static final String SUBSCRIPTION_KEY_NAME = "bhaskar";
+    private static final String DEFAULT_SUBSCRIPTION_URL = "http://appfeed.bhaskar.com/webfeed/apidata/firefox?pageSize=%d&channel_slno=%d&pageNumber=%d";
+    private final String subscriptionUrl;
+    private static final int FIRST_PAGE = 1;
+    private Context context;
+    private int channel;
+    private int pageSize;
+    private String userAgent;
+    private int socketTag;
+    private OnDataChangedListener onDataChangedListener;
+    private OnCacheInvalidateListener onCacheInvalidateListener;
+    private List<ItemPojo> itemPojoList;
+    private boolean cacheIsDirty = false;
+
+    Repository(Context context, int channel, int pageSize, String userAgent, int socketTag, OnDataChangedListener onDataChangedListener, OnCacheInvalidateListener onCacheInvalidateListener, String subscriptionUrl) {
+        this.context = context;
+        this.channel = channel;
+        this.pageSize = pageSize;
+        this.onDataChangedListener = onDataChangedListener;
+        this.onCacheInvalidateListener = onCacheInvalidateListener;
+        this.subscriptionUrl = subscriptionUrl;
+        this.userAgent = userAgent;
+        this.socketTag = socketTag;
+        itemPojoList = new ArrayList<>();
+        nextSubscription();
+    }
+
+    Repository(Context context, int channel, int pageSize, String userAgent, int socketTag, OnDataChangedListener onDataChangedListener, OnCacheInvalidateListener onCacheInvalidateListener) {
+        this(context, channel, pageSize, userAgent, socketTag, onDataChangedListener, onCacheInvalidateListener, DEFAULT_SUBSCRIPTION_URL);
+    }
+
+    public void loadMore() {
+        nextSubscription();
+    }
+
+    // Not used yet and not tested yet.
+    public void reloadData() {
+        itemPojoList = new ArrayList<>();
+        cacheIsDirty = true;
+        nextSubscription();
+    }
+
+    private void nextSubscription() {
+        if (currentPageSubscription != null) {
+            return;
+        }
+        PageSubscription ret = new PageSubscription();
+        ret.page = currentPage;
+        ret.cachedRequestLoader = createLoader(ret.page);
+        currentPage++;
+        ResponseData responseData = ret.cachedRequestLoader.getStringLiveData();
+        ret.observer = createObserver(responseData, ret.page);
+        responseData.observeForever(ret.observer);
```

Comment:
if we reach this line for the second time, the previous observer will still be observed?